app/src/main/java/com/example/student_app_assignment/StudentAdapter.kt

```diff
@@ -0,0 +1,38 @@
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.CheckBox
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.student_app_assignment.R
+import com.example.studentsapp.Student
+
+class StudentAdapter(
+    private val students: List<Student>,
+) : RecyclerView.Adapter<StudentAdapter.StudentViewHolder>() {
+
+    // ViewHolder class to hold references to the views for each item
+    class StudentViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```

Comment:
there is a lot of things raleaated only to create.... remove and put in the right place